### PR TITLE
[IMP] web: allow the use of setting tags on the form view

### PR DESCRIPTION
**Pull request by jpp-odoo**
[IMP] web: documentation link widget

This commit introduces the documentation link widget, the aim is to
standardize the help documentation links that can be found in the views.

This widget will add an icon (o_doc_link), this icon is a link to the
documentation. Note that you can use relative or absolute path. The
relative path is relative to `https://www.odoo.com/documentation/server_version`,
so it's not necessary to hard-code the server version on the arch any more.

```xml
<widget name="documentation_link"  documentation="/applications/technical/web/settings/this_is_a_test.html"/>
```

---
 
[IMP] web: allow the use of setting tags on the form view

The setting tag was introduced in [1], it was restricted to the setting
view (the js_class : base_settings).

This commit move the setting tag to the form view, to remove the
restriction and allow the use of the setting tag in any form view.

[1]: https://github.com/odoo-dev/odoo/commit/c7c2959449effdf0e26f2e2c8e2cdd232a27ca62

---
 
[IMP] *: use the new setting tag on the form view.

*: hr_timesheet, point_of_sale, pos_epson_printer, project,
    sale_project, sale_timesheet

### CODE REVIEW

**Review by aab-odoo — COMMENTED** [review]
looks clean :-)

File: addons/web/static/src/webclient/settings_form_view/settings/setting_header.js [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
-import { Setting } from "./setting";
+import { Setting } from "../../../views/form/setting/setting";
 
 export class SettingHeader extends Setting {
-    setup() {
-        super.setup();
-        // Don't search on a header setting
-        useChildSubEnv({ searchState: { value: "" } });
-    }
-
```
[the remark this answers is not in the document]
Comment (jpp-odoo) [reply addons/web/static/src/webclient/settings_form_view/settings/setting_header.js:12]: Setting (the one in form view) don't have the search part, so we don't need to override this any more.

File: addons/web/static/src/views/widgets/documentation_link/documentation_link.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+DocumentationLink.template = "web.DocumentationLink";
+DocumentationLink.props = {
+    ...standardWidgetProps,
+    record: { type: Object, optional: 1 }, // The record is not needed in this widget
+    documentation: { type: String },
+};
+DocumentationLink.extractProps = ({ attrs }) => {
+    const { documentation } = attrs;
+    return {
+        documentation,
```
Comment (aab-odoo) [inline addons/web/static/src/views/widgets/documentation_link/documentation_link.js:31]: suggestion: rename into path?

File: addons/web/static/src/views/widgets/documentation_link/documentation_link.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.o_widget_documentation_link {
+    display: contents;
```
Comment (aab-odoo) [inline addons/web/static/src/views/widgets/documentation_link/documentation_link.scss:2]: not sure about this. In fields, we can use additionalClasses to add class on the parent node. Would the same feature for widgets allow you to get rid of this rule?

File: addons/web/static/tests/views/form/form_view_tests.js [hunk 4]
```diff
@@ -13042,4 +13042,51 @@ QUnit.module("Views", (hooks) => {
         await click(target.querySelector(".o_data_row .o_data_cell"));
         assert.containsOnce(target, ".o_form_view");
     });
+
+    QUnit.test("documentation_link: relative path", async function (assert) {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form/form_view_tests.js:13118]: I'd add those tests in a document_link_tests.js file, in the widgets/ folder

**Review by aab-odoo — APPROVED** [review]
Besides some small remarks on the documentation widget, this PR looks good :+1:

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Comment by AntoineVDV** [comment]
Very nice addition, in particular the documentation link widget :+1: 
Is it planned to update all views containing an absolute link to the documentation to use the widget instead?

**Comment by jpp-odoo** [comment]
> Very nice addition, in particular the documentation link widget 👍 Is it planned to update all views containing an absolute link to the documentation to use the widget instead?
Hello @AntoineVDV ,

Thanks for your comment, unfortunately for the moment is not planned to update the views to hunt the absolute links (personally I think it can be a good idea).

**Comment by AntoineVDV** [comment]
[That](https://github.com/odoo/odoo/pull/110428) - [would](https://github.com/odoo/odoo/pull/103249) - [**definitely**](https://github.com/odoo/odoo/pull/93266) - [be a](https://github.com/odoo/odoo/pull/88814) - [good](https://github.com/odoo/odoo/pull/84445) - [idea](https://github.com/odoo/odoo/pull/77872) :sweat_smile: 
cc @bouvyd

**Pull request closed without merging**